tofhir-log-server/src/main/scala/io/tofhir/server/service/ExecutionService.scala

```diff
@@ -150,17 +162,32 @@ class ExecutionService() extends LazyLogging {
           val page = queryParams.getOrElse("page", "1").toInt
           // handle the case where requested page does not exist
           if (page > numOfPages) {
-            (Seq.empty, 0)
+            (Seq.empty, 0, 0)
           } else {
             val start = (page - 1) * 10
             val end = Math.min(start + 10, total.toInt)
+
+            var filteredLogs = executionLogs
+            // Filter according to error status of the execution
+            if(Option(errorStatuses).nonEmpty){
+              filteredLogs = filteredLogs.filter(col("errorStatus").isin(errorStatuses.split(","): _*))
+            }
+
+            // Filter according to start date
+            if(Option(dateAfter).nonEmpty){
+              filteredLogs = filteredLogs.filter(col("startTime") > dateAfter)
+            }
+            if (Option(dateBefore).nonEmpty) {
+              filteredLogs = filteredLogs.filter(col("startTime") < dateBefore)
+            }
+
             // sort the executions by latest to oldest
-            val paginatedLogs = executionLogs.sort(executionLogs.col("startTime").desc).collect().slice(start, end)
+            val paginatedLogs = filteredLogs.sort(executionLogs.col("startTime").desc).collect().slice(start, end)
```

Comment:
**start** and **end** parameters are calculated based on the **total** variable which is wrong because we need to find these variables after the filtering is done. 

Further, please update the tests to reproduce this error. To do this, you can set page size to 4 and simply pass "page" -> "2" parameter to "should get executions filtered by a date range" test. 

In addition to this bug, I think it would be great if you can update the function so that it takes the page size as a query parameter whose default value is 10 which is hardcoded now.